model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt

```diff
@@ -328,6 +336,34 @@ class ModelClientV2(
         }
     }
 
+    private suspend fun createVersion(baseVersion: CLVersion?, delta: Flow<String>): CLVersion {
+        var firstHash: String? = null
+        var isHash = true
+        var lastHash: String? = null
+        delta.collect {
+            if (isHash) {
+                lastHash = it
+                if (firstHash == null) {
+                    firstHash = it
+                }
+            } else {
+                val value = it
+                store.keyValueStore.put(lastHash!!, value)
+            }
+            isHash = !isHash
```

Comment:
Is `windowed` available on flows? The mutability and the swapping of the boolean can then maybe be expressed in a way that is a bit easier to read.

Reply:
No, it's not.

Reply:
Not creating a new list for each key/value pair (as windowed does) is probably also relevant for the performance.

Reply:
No idea how much the compiler could/would optimize here. But as it's not available, there's nothing to be done here.